pkg/operator/controller/ingress/deployment_test.go

```diff
@@ -609,17 +670,14 @@ func TestDesiredRouterDeployment(t *testing.T) {
 		},
 		Tolerations: []corev1.Toleration{toleration},
 	}
-	expectedReplicas = 3
+	expectedReplicas := int32(3)
 	ci.Spec.Replicas = &expectedReplicas
 	ci.Status.EndpointPublishingStrategy.Type = operatorv1.HostNetworkStrategyType
 	networkConfig.Status.ClusterNetwork = []configv1.ClusterNetworkEntry{
 		{CIDR: "2620:0:2d0:200::7/32"},
 	}
-	proxyNeeded, err = IsProxyProtocolNeeded(ci, infraConfig.Status.PlatformStatus)
-	if err != nil {
-		t.Errorf("failed to determine infrastructure platform status for ingresscontroller %s/%s: %v", ci.Namespace, ci.Name, err)
-	}
```

Comment:
It isn't clear why you deleted the recomputation of `proxyNeeded` here and not in `TestDesiredRouterDeploymentSpecAndNetwork`.  In both cases, the endpoint publishing strategy changed, so it makes sense to recompute `proxyNeeded`.

Reply:
Fixed.